assets/js/googlesitekit/datastore/site/notifications.test.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * core/site data store: notifications.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { actions } from './index';
+import {
+	ADD_NOTIFICATION,
+	REMOVE_NOTIFICATION,
+	FETCH_NOTIFICATIONS,
+	RECEIVE_NOTIFICATIONS,
+	RECEIVE_NOTIFICATIONS_FAILED,
+} from '../../data/create-notifications-store';
+
+describe( 'core/site notifications', () => {
+	it( 'has appropriate notification methods', () => {
+		const actionsToExpect = [
+			'addNotification',
+			'removeNotification',
+			'fetchNotifications',
+			'receiveNotifications',
+			'receiveNotificationsFailed',
+		];
+		expect( Object.keys( actions ) ).toEqual( expect.arrayContaining( actionsToExpect ) );
+	} );
+
+	it( 'Dispatches addNotification', () => {
+		const notification = {};
+		const expectedAction = {
+			payload: { notification },
+			type: ADD_NOTIFICATION,
+		};
+		expect( actions.addNotification( notification ) ).toEqual( expectedAction );
```

Comment:
So, I get the intention of checking the actions here but again: if the internal implementation of how notifications dispatch actions changes we'll have to update the tests here even though they aren't related. This is testing how notifications dispatch their actions instead of _that_ our store has notifications. If we change the `addNotification` action creator by adding a new attribute to the payload, it shouldn't affect this datastore as the notification part of the store will handle things.

But we'd have to update this test even though nothing about this store or the functions it used would change externally. So I think it's best to just have the `has appropriate notification methods` test for this file; testing the individual actions and especially their payloads (which are not consumed by reducers outside of the notifications store—they're essentially "private"/internal) is fragile and doesn't provide much extra benefit to this store.

I think it's best to remove these and just leave the "has appropriate notification methods" one. Sorry if my previous suggestion was confusing!

Reply:
@tofumatt I've removed the tests. Thanks for your feedback!